### PR TITLE
Upgrade to IPython 7

### PR DESCRIPTION
**Pull request by ulope**
## Description

Merge-after: #6189 

We were stuck on ancient IPython 4.x

This also required to upgrade out ancient sphinx version but it seems
there are no incompatibilities caused by it.

### CODE REVIEW

File: raiden/ui/console.py [hunk 1]
```diff
@@ -37,13 +36,10 @@
 OKGREEN = "\033[92m"
 ENDC = "\033[0m"
 
-# ipython needs to accept "--gui gevent" option
-IPython.core.shellapp.InteractiveShellApp.gui.values += ("gevent",)
-
 
 def print_usage() -> None:
+    print(f"\tuse `{HEADER}app{OKBLUE}` to interact with the top level Raiden App API.")
```
Comment (palango) [inline raiden/ui/console.py:41]: Nice catch 👀

File: raiden/ui/console.py [hunk 2]
```diff
@@ -37,13 +36,10 @@
 OKGREEN = "\033[92m"
 ENDC = "\033[0m"
 
-# ipython needs to accept "--gui gevent" option
-IPython.core.shellapp.InteractiveShellApp.gui.values += ("gevent",)
```
Comment (palango) [inline raiden/ui/console.py:41]: This works without all that gevent stuff now?

File: raiden/ui/console.py [hunk 3]
```diff
@@ -153,7 +113,7 @@ def lasterr(n: int = 1) -> None:
         print("Entering Console" + OKGREEN)
         print("Tip:" + OKBLUE)
         print_usage()
-        IPython.start_ipython(argv=["--gui", "gevent"], user_ns=self.console_locals)
+        IPython.start_ipython(argv=[], user_ns=self.console_locals)
```
Comment (konradkonrad) [inline raiden/ui/console.py:116]: nice! ipython now has native gevent support?

File: raiden/ui/console.py (hunk 2 above)
Comment (ulope) [reply raiden/ui/console.py:41]: Yes, from 5.0 onwards IPython stopped using gnureadline (which was the point of all this) and instead uses prompt-toolkit which is pure python. Therefore it seems that gevent monkeypatching makes it work.

File: raiden/ui/console.py (hunk 3 above)
Comment (ulope) [reply raiden/ui/console.py:116]: No, but see above. Due to monkeypatching it seems to work automatically.

**Pull request merged**